science/bali-phy/Portfile

```diff
@@ -1,110 +1,39 @@
 PortSystem 1.0
-
+#PortGroup           github 1.0
 name                bali-phy
-version             2.1.1
-revision            5
-description         Bayesian Alignment and Phylogeny estimation
-long_description    BAli-Phy can estimate phylogenetic trees from sequence data when the alignment \
-                    is uncertain. Instead of conditioning on a single alignment estimate, BAli-Phy \
-                    accounts for alignment uncertainty by integrating over all alignments. \
-                    BAli-Phy does not rely on a guide tree because the alignment and the tree are \
-                    co-estimated. Therefore it can construct phylogeny estimates of widely \
-                    divergent sequences without bias toward a guide tree.
+version             3.0-beta6-20171223
+revision            1
+description         Estimation of phylogeny and multiple alignment via MCMC
+long_description    Estimation of phylogeny and multiple alignment via MCMC \
+                    BAli-Phy that estimates multiple sequence alignments and \
+                    evolutionary trees from DNA, amino acid, or codon sequences.\
+                    BAli-Phy uses MCMC and Bayesian  methods to estimate \
+                    evolutionary trees, positive selection, and branch lengths \
+                    while averaging over alternative alignments. \
+                     \
+                    BAli-Phy can also estimate phylogenies from a fixed \
+                    alignment (like MrBayes and BEAST) using substitution \
+                    models like GTR+gamma.  BAli-Phy automatically estimates \
+                    relative rates for each gene.
 categories          science
 platforms           darwin
-maintainers         nomaintainer
-homepage            http://www.biomath.ucla.edu/msuchard/bali-phy/
+license             GPL-2
+maintainers         gmail.com:benjamin.redelings openmaintainer
+homepage            http://www.bali-phy.org/
 
-master_sites        ${homepage}
-checksums           sha1 e72073a1c5b05c797668e476bfd8517594f074e6
+# By using a SHA, this should be completely reproduceable.
+fetch.type          git
+git.url             https://github.com/bredelings/BAli-Phy.git
+git.branch          217244e8dab7019d2501763ad0f97b8ef7473018
 
-patchfiles          configure.patch scripts.patch src-tools-draw-tree.patch
+depends_build       port:pkgconfig port:automake port:autoconf port:libtool
+depends_lib         port:boost path:lib/pkgconfig/cairo.pc:cairo
 
-depends_lib         port:gsl port:boost path:lib/pkgconfig/cairo.pc:cairo
-depends_build       port:pkgconfig
+pre-configure       { system "cd ${workpath} &&  mkdir -p build && cd ${worksrcpath} && ./bootstrap.sh"}
```

Comment:
Wait ... does that mean that you just gave up the idea of using meson?
I just wanted to say that in case you need a patch in meson, we can patch meson (before your patches make it into an official release) for the sake of being able to find boost. Lots of these complications could go. But even if you don't use meson, I assume that quite a bit of this code could be simplified.

Reply:
Hmm... I haven't given up.  If we can patch meson, then I think it will work.  I'm thinking of a two-phase approach where I go from BROKEN -> WORKING/autotools -> WORKING/meson.  I just wanted it to actually build!

I suppose this should be able to be simplified, since all I want to do is:
* run a script in the source directory
* use a separate build directory
Can you point me to a portfile that does this right?